lib/src/dropdown/Dropdown.tsx

```diff
@@ -20,145 +29,190 @@ const DxcDropdown = ({
-          >
-            <DropdownTriggerContainer caretHidden={caretHidden}>
-              {iconPosition === "after" && labelComponent}
-              {icon && (
-                <ButtonIconContainer label={label} iconPosition={iconPosition} disabled={disabled}>
-                  {typeof icon === "string" ? <ButtonIcon src={icon} /> : icon}
-                </ButtonIconContainer>
-              )}
-              {iconPosition === "before" && labelComponent}
-            </DropdownTriggerContainer>
-            <CaretIconContainer caretHidden={caretHidden} disabled={disabled}>
-              {!caretHidden && (anchorEl === null ? <DownArrowIcon /> : <UpArrowIcon />)}
-            </CaretIconContainer>
-          </DropdownTrigger>
-          <DXCMenu
-            anchorEl={anchorEl}
-            open={Boolean(anchorEl)}
-            onClose={handleClose}
-            getContentAnchorEl={null}
-            anchorOrigin={{ vertical: "bottom", horizontal: "left" }}
-            transformOrigin={{ vertical: "top", horizontal: "left" }}
-            size={size}
-            width={width}
-            role={undefined}
-            transition
-            disablePortal
-            placement="bottom-start"
-          >
-            {({ TransitionProps }) => (
-              <Grow {...TransitionProps}>
-                <Paper>
-                  <ClickAwayListener onClickAway={handleClose}>
-                    <MenuList autoFocusItem={Boolean(anchorEl)} id="menu-list-grow">
-                      {options.map((option) => (
-                        <MenuItem
-                          key={option.value}
-                          value={option.value}
-                          disableRipple={true}
-                          onClick={(event) => handleMenuItemClick(option)}
-                        >
-                          {optionsIconPosition === "after" && <span className="optionLabel">{option.label}</span>}
-                          {option.icon && (
-                            <ListIconContainer label={option.label} iconPosition={optionsIconPosition}>
-                              {typeof option.icon === "string" ? <ListIcon src={option.icon} /> : option.icon}
-                            </ListIconContainer>
-                          )}
-                          {optionsIconPosition === "before" && <span className="optionLabel">{option.label}</span>}
-                        </MenuItem>
-                      ))}
-                    </MenuList>
-                  </ClickAwayListener>
-                </Paper>
-              </Grow>
-            )}
-          </DXCMenu>
-        </div>
-      </DXCDropdownContainer>
+      <DropdownContainer
+        onMouseEnter={!disabled && expandOnHover ? handleOnOpenMenu : undefined}
+        onMouseLeave={!disabled && expandOnHover ? handleOnCloseMenu : undefined}
+        onBlur={!disabled ? handleOnBlur : undefined}
+        margin={margin}
+        size={size}
+      >
```

Comment:
I tried before and It didn't work as intended, let me try again.

Reply:
Yes, I confirm. This cannot be achieved for some reasons:

1. The focus here is a little bit trickier than in other components such as Select or Autosuggest. There, the focus always stays on the trigger that opens the list, but in the dropdown that is not correct. The focus moves between the list and the button when opening and closing, and should go out when clicked outside of them. Since we are using portals for the menu this becomes more difficult because the menu is not inside the main container, so I needed to play with the events, their propagation, etc. I need the outside container so the trigger stops its onBlur propagation (when the button loses the focus to the menu it should not close it).
2. `expandOnHover` is another issue here. If we land the `mouseEnter` and `mouseLeave` events in the trigger instead of the container the bubbling would close the menu when the cursor goes from the trigger to the menu.

Reply:
Okay! Nice explanation Ivan! 💯